### ISSUE TITLE
search for tokens in `style` attr only

### ISSUE THREAD

**Issue by jjenzz** [tokenami/tokenami#49]
currently we search for tokens anywhere in a file so if someone talks about `---bg-color` in some copy, we will generate the style output for it.

we should search `style` attr only and allow a `tokenRegex` in `tokenami.config.js` so consumers can configure other regexes to search in files.

**Comment by ubmit** [comment]
@jjenzz searching for tokens only in the `style` attr wouldn't work for cases such as this one I guess

```
import { css } from '@tokenami/css';

function Button({ size, style, ...props }) {
  return <button {...props} style={button({ size }, style)} />;
}

const button = css(
  { '---padding': 4 },
  {
    size: {
      small: { '--padding': 2 },
      large: { '--padding': 6 },
    },
  },
  { responsive: true }
);
```

Maybe a good default would be looking for any function call inside the `style` attr first. If there is one, we could check for the tokens inside the fn definition. If not, we can simply check the `style` attr value. What do you think?

**Comment by jjenzz** [comment]
ah sorry @ubmit i wrote this ticket long before the `css` utility existed and forgot to update it. you are correct, we need to search inside `style` and `css({})` and also allow any additional regexes to be configured in the `tokenami.config`.

**Comment by pawelblaszczyk5** [comment]
My two cents on that issue - I’m wondering whether that’s something we really want 🤔 Currently I can just create an object, slap some `TokenamiProperties` for intellisense and I’ve created reusable styles. I don’t need an identity function that’s specifically for some regex to catch my styles. I can even store a selector in variable to use it later on. And the only downside that I’m aware of is that extractor can create a small bit of unused CSS. I don’t think it’s too common to produce code that would bloat the CSS so I see it as a non-issue. 

Tailwind also parse all the source code for extraction (but in its case you need to use regexes and stuff for intellisense anyway).

**Comment by jjenzz** [comment]
@pawelblaszczyk5 yeah, that's a fair point. the `tokenami.config` already has an `exclude` option if ppl want to exclude markdown files or something from generating styles (say they're authoring some docs).

@ubmit perhaps we should park this one for now cos i've also more recently been thinking that the way it currently works isn't that big of an issue.

**Comment by ubmit** [comment]
> @pawelblaszczyk5 yeah, that's a fair point. the `tokenami.config` already has an `exclude` option if ppl want to exclude markdown files or something from generating styles (say they're authoring some docs).
> 
> 
> 
> @ubmit perhaps we should park this one for now cos i've also been thinking that the way it currently works isn't that big of an issue.

Got it 🫡
